samples/Test.AppWriter/ExampleAppGenerator.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//using System;
+//using System.Collections.Generic;
+//using System.Linq;
+//using System.Text;
+//using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.PowerPlatform.PowerApps.Persistence.Templates;
+using Microsoft.PowerPlatform.PowerApps.Persistence.Models;
+
+namespace Test.AppWriter;
+internal class ExampleAppGenerator
+{
+    public static List<(string, string)> ParseControlsInfo(string[] controlsinfo)
+    {
+        var tuples = new List<(string, string)>();
+
+        for (var i = 0; i < controlsinfo.Length; i += 2)
+        {
+            tuples.Add((controlsinfo[i], controlsinfo[i + 1]));
+        }
+
+        return tuples;
```

Comment:
Style opinion: LINQ is often more concise and readable.  
(Readability of course being subjective, but I find it simpler ¯\\\_(ツ)_/¯ )

```suggestion
        return controlsinfo
            .Chunk(2)
            .Select(pair => (pair[0], pair[1]))
            .ToList();
```

Reply:
The structure in which this specific data will be gathered from the commandline is still a work in progress, but stylistically I have no complaints about using the suggested format.